src/Sulu/Component/Content/SmartContent/PageDataProvider.php

```diff
@@ -400,6 +419,28 @@ function(
         );
     }
 
+    private function getTypes(): array
+    {
+        $types = [];
+        if ($this->tokenStorage && null != $this->tokenStorage->getToken()) {
+            $user = $this->tokenStorage->getToken()->getUser();
+
+            // user may be "anon."
+            if (!$user instanceof UserInterface) {
```

Comment:
Is that a real possibility? Because if this should not happen, I would throw an exception in this `if`. Would make it easier to spot errors later on.

Reply:
The `PageDataProvider` configuration is initialized on the first `/admin` request. If the user is not logged in, then the `user` is "anon."

Reply:
Are you saying that the `initConfiguration` function is also called if the user is not logged in? I thought the `/config` request is only sent if the user has been logged in :thinking:

Reply:
Yes, when the `AudienceTargeting` is not active, the configuration for every `dataProvider` is initialized.
See: https://github.com/sulu/sulu/blob/master/src/Sulu/Component/SmartContent/DataProviderPool.php#L43-L45